### PR TITLE
refactor: remove consistent-return rule

### PR DESCRIPTION
**Pull request by kieckhafer**
The `consistent-return` rule is causing some confusion as to how it is implemented, and doesn't seem to be enforcing what we want it to enforce.

For example, take this function:
```
getCurrentTag() {
    if (this.Router.getRouteName() === "tag") {
      return this.Router.current().params.slug;
    }
  }
```

Even with the option `` enabled, this throws an error. When we try and return `return` or `return undefined, which is what we need this function to return, the error is still thrown.

It seems the rule requires a same-type return, for example a `null` or `string` to be returned, instead of `undefined`.

Because of this confusion, combined with this rule not being one of the `eslint:recommended` rules ([see here](https://eslint.org/docs/rules/)), we are going to remove this rule.

### CODE REVIEW

**Comment by kieckhafer** [comment]
Closing in favor of #17

**Comment by kieckhafer** [comment]
After some discussion, seems we'd like to keep this rule. Closing.

**Comment by spencern** [comment]
@kieckhafer where was the discussion? Can you summarize?

**Comment by kieckhafer** [comment]
Basically we (collectively) were interpreting the incorrectly and thought it was doing the wrong thing, but in the end, it was doing the right thing and we misunderstood what it was supposed to be doing.
